numba/parfors/parfor_lowering_utils.py

```diff
@@ -0,0 +1,211 @@
+
+
+class ParforLoweringBuilder:
+    def __init__(self, lowerer, scope, loc):
+        self._lowerer = lowerer
+        self._scope = scope
+        self._loc = loc
+
+    @property
+    def _context(self):
+        return self._lowerer.context
+
+    @property
+    def _typingctx(self):
+        return self._context.typing_context
+
+    @property
+    def _typemap(self):
+        return self._lowerer.fndesc.typemap
+
+    @property
+    def _calltypes(self):
+        return self._lowerer.fndesc.calltypes
+
+    def bind_global_function(self, fobj, ftype, args, kws={}):
+        """Bind a global function to a variable.
+
+        Parameters
+        ----------
+        fobj : object
+            The function to be bounded.
+        ftype : types.Type
+        args : Sequence[types.Type]
+        kws : Mapping[str, types.Type]
+
+        Returns
+        -------
+        callable: _CallableNode
+        """
+        loc = self._loc
+        varname = f"{fobj.__name__}_func"
+        gvname = f"{fobj.__name__}"
+
+        func_sig = self._typingctx.resolve_function_type(ftype, args, kws)
+        func_var = self.assign(
+            rhs=ir.Global(gvname, fobj, loc=loc), typ=ftype, name=varname
+        )
+        return _CallableNode(func=func_var, sig=func_sig)
+
+    def make_const_variable(self, cval, typ, name="pf_const") -> ir.Var:
+        """Make a constant variable
+
+        Parameters
+        ----------
+        cval : object
+            The constant value
+        typ : types.Type
+            type of the value
+        name : str
+            variable name to store to
+
+        Returns
+        -------
+        res : ir.Var
+        """
+        return self.assign(
+            rhs=ir.Const(cval, loc=self._loc), typ=typ, name=name
+        )
+
+    def make_tuple_variable(self, varlist, name="pf_tuple") -> ir.Var:
+        """Make a tuple variable
+
+        Parameters
+        ----------
+        varlist : Sequence[ir.Var]
+            Variables containing the values to be stored.
+        name : str
+            variable name to store to
+
+        Returns
+        -------
+        res : ir.Var
+        """
+        loc = self._loc
+        vartys = [self._typemap[x.name] for x in varlist]
+        tupty = types.Tuple.from_types(vartys)
+        return self.assign(
+            rhs=ir.Expr.build_tuple(varlist, loc), typ=tupty, name=name
+        )
+
+    def assign(self, rhs, typ, name="pf_assign") -> ir.Var:
+        """Assign a value to a new variable
+
+        Parameters
+        ----------
+        rhs : object
+            The value
+        typ : types.Type
+            type of the value
+        name : str
+            variable name to store to
+
+        Returns
+        -------
+        res : ir.Var
+        """
+        loc = self._loc
+        var = ir.Var(self._scope, mk_unique_var(name), loc)
+        self._typemap[var.name] = typ
+        redshape_assign = ir.Assign(rhs, var, loc)
```

Comment:
```suggestion
        assign = ir.Assign(rhs, var, loc)
```
don't think this has anything to do with reduction?